src/graphics/mesh.rs

```diff
@@ -230,6 +232,26 @@ impl MeshBuilder {
     ) -> GameResult<&mut Self>
     where
         P: Into<mint::Point2<f32>> + Clone,
+    {
+        let vb = VertexBuilder {
+            color: LinearColor::from(color),
+        };
+        self.polyline_with_vertex_builder(mode, points, is_closed, vb)
+    }
+
+    /// Create a new mesh for a given polyline using a custom vertex builder.
+    /// The points given must be in clockwise order.
+    pub fn polyline_with_vertex_builder<P, V>(
+        &mut self,
+        mode: DrawMode,
+        points: &[P],
+        is_closed: bool,
+        vb: V,
+    ) -> GameResult<&mut Self>
+    where
+        P: Into<mint::Point2<f32>> + Clone,
+        V: t::VertexConstructor<t::FillVertex, Vertex>
+            + t::VertexConstructor<t::StrokeVertex, Vertex>,
```

Comment:
Exposing `lyon`'s `VertexConstructor` types publicly here are probably not a good idea, since we don't re-export `lyon` as a whole.

Reply:
I've added this to be able to provide a custom vertex builder to build polylines which in turn were needed to generate proper UVs and vertex colors and currently there's no reasonable way of doing that. It is possible to iterate over vertices after the mesh was built, but this would over complicate the code and dramatically reduce performance.

I agree that this is probably not a good idea, I'd love to discuss this issue further.

Reply:
Lyon 0.14 has them removed in favor of regular closures, maybe upgrading it will solve this issue?

Reply:
That's an interesting idea, I'll have to try...